### PR TITLE
feat: account deletion flow

### PR DESCRIPTION
**Pull request by LazyAfternoons**
## Short description
This PR adds the profile deletion flow which can be invoked from any point of the application. In this case, it can be started from the profile page by interacting with the "profile deletion status" switch or from the wallet by pressing the "delete profile" button.

## List of changes proposed in this pull request
- locales/en/index.yml & locales/it/index.yml: Adds locale strings for the deletion flow;
- ts/components/wallet/WalletHomeHeader.tsx: Adds a button to invoke the profile deletion flow;
- ts/navigation/ProfileNavigator.tsx: Adds the profile deletion flow screens;
- ts/navigation/params/ProfileParamsList.ts: Adds the profile deletion flow screens parameters;
- ts/navigation/routes.ts: Adds the profile deletion flow screens routes;
- ts/screens/profile/NewProfileScreen.tsx: Adds the `onChange` prop to the switch to invoke the profile deletion flow and refactors the usage of `getValueFromProfilePot`;
- ts/screens/profile/ProfileDeletionDetailsScreen.tsx: Adds a screen which shows details of the data which is going to be deleted along with the profile and again two buttons, the first one to cancel the operation and the second one to continue;
This screen also features a loading view, an error view with a `retry` button and a continue view which confirms that the request is being processed and closes the flow.;
- ts/screens/profile/ProfileDeletionScreen.tsx: Adds a screen which shows a warning for the user and two buttons, the first one to cancel the operation and the second one to continue;
- ts/utils/pot.ts: Moves the `getValueFromProfilePot` function here.

## How to test
- Point to the dev server;
- Generate locales via yarn generate, then start the app and navigate to Profile through the bottom navigator;
- Toggle the switch status to start the flow;
- The first screen should show an `InfoBox` and two buttons as explained above. The cancel button should make you return to the profile screen; 
- The second screen should show the details of the deletion and two buttons, as explained above. The cancel button should make you return to the profile screen;
- If everything goes as it should a new screen with a continue button should appear. The `continue` button should make you return to the profile screen; 
- The same applies when starting the flow from the wallet screen by pressing on the `Delete account` button.

### CODE REVIEW

File: ts/screens/profile/NewProfileScreen.tsx [hunk 1]
```diff
@@ -113,21 +109,22 @@ const NewProfileScreen = () => {
...
+            newProfilePot,
+            "name"
+          )} ${getValueFromProfilePot(newProfilePot, "family_name")}`}
           hideIcon
           leadingIcon={nameSurnameIcon}
         ></ListItemComponent>
         <ListItemComponent
           title={I18n.t("profile.data.list.fiscalCode")}
-          subTitle={getValueFromPot("fiscal_code")}
+          subTitle={getValueFromProfilePot(newProfilePot, "fiscal_code")}
```
Comment (shadowsheep1) [inline ts/screens/profile/NewProfileScreen.tsx:121]: Does `tsc --noemit` complain? ;)

File: ts/screens/profile/NewProfileScreen.tsx [hunk 2]
```diff
@@ -113,21 +109,22 @@ const NewProfileScreen = () => {
...
           hideIcon
           leadingIcon={cardIcon}
         ></ListItemComponent>
         <ListItemComponent
           title={I18n.t("profile.data.list.email")}
-          subTitle={getValueFromPot("email")}
+          subTitle={getValueFromProfilePot(newProfilePot, "email")}
```
Comment (shadowsheep1) [inline ts/screens/profile/NewProfileScreen.tsx:127]: as above

File: ts/screens/profile/ProfileDeletionDetailsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    if (isLoading) {
+      return <LoadingView />;
+    } else if (isError) {
+      return <ErrorView />;
+    } else if (isRequestPending) {
+      return <ContinueView />;
+    } else {
+      return <ContentView />;
+    }
+  };
```
Comment (shadowsheep1) [inline ts/screens/profile/ProfileDeletionDetailsScreen.tsx:172]: What do you think to use a more functional approach? :)

File: ts/screens/profile/NewProfileScreen.tsx (hunk 1 above)
Comment (LazyAfternoons) [reply ts/screens/profile/NewProfileScreen.tsx:121]: My bad, I forgot a cast when i re-wrote the function with a functional approach. Fixed in #[8e56640](https://github.com/LazyAfternoons/io-app/pull/4#:~:text=cast%20in%20getValueFromProfilePot-,8e56640,-Merge%20state)

File: ts/screens/profile/ProfileDeletionDetailsScreen.tsx (hunk 3 above)
Comment (LazyAfternoons) [reply ts/screens/profile/ProfileDeletionDetailsScreen.tsx:172]: I think it's way less readable in this case, or at least with what I had in mind. What do you think?
```
return pipe(
      isLoading,
      B.fold(
        () =>
          pipe(
            isError,
            B.fold(
              () =>
                pipe(
                  isRequestPending,
                  B.fold(
                    () => <ContinueView />,
                    () => <ContentView />
                  )
                ),
              () => <ErrorView />
            )
          ),
        () => <LoadingView />
      )
    );
```

Comment (shadowsheep1) [reply ts/screens/profile/ProfileDeletionDetailsScreen.tsx:172]: Yep that true. But what about something like that?

```typescript
return isLoading ? (
      <LoadingView />
    ) : isError ? (
      <ErrorView />
    ) : isRequestPending ? (
      <ContinueView />
    ) : (
      <ContentView />
    );
```

What do you think?

Comment (LazyAfternoons) [reply ts/screens/profile/ProfileDeletionDetailsScreen.tsx]: I really like this one, implemented in [47fa232](https://github.com/LazyAfternoons/io-app/pull/4/commits/47fa232e40f52367f1e8f68f56fc7b3312b96696). Let me know if this is safe and sound, thank you!

Comment (shadowsheep1) [reply ts/screens/profile/ProfileDeletionDetailsScreen.tsx]: Safe and sound 🚀

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**